src/utils.js

```diff
@@ -63,16 +65,86 @@ async function checkForPlaystationDirectRedirect(checkInterval, onSuccess, versi
         onSuccess();
     } else {
         setTimeout(() => {
-            console.log("No stock detected yet. Scanning again...");
-            console.log("Number of tries", numTries);
-            console.log("");
+            console.log("PS Direct: No queue detected yet. Scanning again...");
+            console.log(`Number of tries: ${numTries}\n`);
             numTries++;
 
             checkForPlaystationDirectRedirect(checkInterval, onSuccess, version, browser, numTries);
         }, checkInterval);
     }
 }
 
+/*  Recursively checks walmart mobile API for online and inStore stock status
+    storeId param based on user input -- set to 1 if no user input, will only
+    check online stock
+*/
+
+async function watchWalmart(checkInterval, onWalmartSuccess, version, storeId, numWalmartTries = 1) {
+    let walmartStockStatus = await checkWalmartStock(version, storeId);
+    let walmartRetry;
+
+    switch(walmartStockStatus) {
+        case "walmartConnectionError":
+            console.log("Walmart Connection Error");
+            console.log(`Number of tries: ${numWalmartTries}\n`);
+            walmartRetry = true;
+            break;
+        case "walmartOnlineFound":
+            console.log("Found Walmart Online Inventory!");
+            onWalmartSuccess();
+            break;
+        case "walmartInStoreFound":
+            console.log("Found Walmart Store Pickup Inventory!");
+            onWalmartSuccess();
+            break;
+        case "noPage":
+            console.log("Product page not up yet.");
+            console.log(`Walmart Number of tries: ${numWalmartTries}`);
+            walmartRetry = true;
+            break;
+        default:
+            console.log("Walmart: No inventory detected yet.  Scanning again...");
+            console.log(`Walmart Number of tries: ${numWalmartTries}\n`);
+            walmartRetry = true;
+            break;
+    }
+    if (walmartRetry == true) {
```

Comment:
You never have to explicitly check if something evaluates to `true` in an if statement. You can simply use:
```
if (warlmartRetry) { ... }
```

Reply:
Yep, aware, but I occasionally like to explicitly check for code clarity for people who don't know much js

Reply:
Gotcha. Well, it's still unnecessary. Please remove.

Reply:
I'd rather have clean code than cater to an audience who likely won't spend much time actually looking through these files.

Reply:
There's also value to explicitly declaring "true" vs if (walmartRetry) in that the latter doesn't *just* look for true, it looks for any truthy value.  Explicit declaration forces the condition to only occur if it's explicitly set to true

Reply:
Just remove == true lol, no need to have it in there

It would never be any truthy value unless it was actually true

Reply:
> Just remove == true lol, no need to have it in there
> 
> It would never be any truthy value unless it was actually true

That's making an assumption that won't necessarily be true as more people start contributing, or when people modify their own code.

That block isn't intended to run if the variable is truthy, only if it's explicitly true.  Making an assumption that "eh, it should only be truthy if it's true" is logic that leads to technical debt downstream.

Reply:
![image](https://user-images.githubusercontent.com/18370646/102265002-47463a00-3ecb-11eb-86dd-9e88761cf7fe.png)
I'm not seeing any value here.

Reply:
Also if your reason behind this was to test for explicit declaration, you would need to use `walmartRetry === true`. Still, this is unnecessary.

Circling back to your assertion that this is meant for non-technical comprehension, consider the following English example:

"If I want to check Walmart again, I will go to their website."

vs

"If I want check Walmart again is true, I will go to their website."

I can't imagine a scenario where someone would say something like the second example, so I believe changing this to `if (walmartRetry)` makes sense. If you're concerned about non-technical comprehension, I would encourage you to change the original variable declaration instead, perhaps to something like `shouldCheckWalmartAgain` or something like that.

Reply:
You could add a setter and getter if you're worried about people setting walmartRetry to a non `true` value :) `shouldRetryWalmart()` and `setRetryWalmart()` which would only set it to a true value